storage/opsmgr.go

```diff
@@ -132,15 +133,26 @@ func newOmAccessor(conf OmConfig) (*OmAccessor, omAuthMetadata, error) {
 //Get attempts to get the secret stored at the requested backend path and
 // return it as a map.
 func (v *OmAccessor) Get(path string) (map[string]string, error) {
+	pathDetails := strings.Split(path, "/")
+	if len(pathDetails) != 2 {
+		return map[string]string{}, fmt.Errorf("path '%s' does not properly specify a product guid and property reference", path)
+	}
+
 	var credentials struct {
 		Cred struct {
 			Type  string            `json:"type"`
 			Value map[string]string `json:"value"`
 		} `json:"credential"`
 	}
 
-	respBody, err := v.opsmanAPI(path)
+	respBody, err := v.opsmanAPI(fmt.Sprintf("/api/v0/deployed/products/%s/credentials/%s", pathDetails[0], pathDetails[1]))
 	if err != nil {
+		if pathDetails[0] == "ops_manager" {
+			return v.opsmanRootFallback(pathDetails[0], pathDetails[1])
+		}
+		if strings.Contains(pathDetails[0], "p-bosh-") {
```

Comment:
Unsure of the path format here, but can this be changed to the more efficient `strings.HasPrefix` instead of Contains?

Reply:
Absolutely - done